src/main/java/com/altenheim/kalender/controller/logicController/WebsiteScraperController.java

```diff
@@ -1,27 +1,70 @@
-package com.altenheim.kalender.controller.logicController;
-
-import java.util.Timer;
-import java.util.TimerTask;
-import com.altenheim.kalender.interfaces.IWebsiteScraperController;
-import com.altenheim.kalender.models.SettingsModel;
-
-public class WebsiteScraperController extends TimerTask implements IWebsiteScraperController
-{
-    private SettingsModel settings;
-
-    public WebsiteScraperController(SettingsModel settings)
-    {
-        this.settings = settings;        
-    }
-
-    public void startScraperTask()
-    {
-        var timer = new Timer();  
-        timer.schedule(this, 0, settings.getScrapingInterval());
-    }
-
-    public void run() 
-    {
-        System.out.println("Scraping...");      
-    }
-}    
\ No newline at end of file
+package com.altenheim.kalender.controller.logicController;
+
+import com.altenheim.kalender.interfaces.IWebsiteScraperController;
+import com.altenheim.kalender.models.SettingsModel;
+import net.fortuna.ical4j.data.ParserException;
+import java.io.BufferedReader;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.ParseException;
+import java.util.Timer;
+    import java.util.TimerTask;
+
+
+    public class WebsiteScraperController extends TimerTask implements IWebsiteScraperController
+    {
+        private SettingsModel settings;
+
+        public WebsiteScraperController(SettingsModel settings)
+        {
+            this.settings = settings;        
+        }
+
+        public void startScraperTask()
+        {
+            var timer = new Timer();  
+            timer.schedule(this, 0, settings.getScrapingInterval());
+        }
+
+        public void run() 
+        {
+            System.out.println("Scraping...");  
+            try {
+				HWRCalendarICS();
+			} catch (IOException | ParseException | ParserException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+        }
+
+		
+		public void HWRCalendarICS() throws IOException, ParseException, ParserException {
+			URL url = new URL(settings.geturlvariables());
+			String line = "";
+			downloadIcs(url, line);
+			
+			//Übergabe String path an Import-Funktion
+			Path ics = Paths.get("hwrCalendar.ics");

```

Comment:
Pfade nicht hardcode, auch nicht relative. Macht ne prop und nen getter in den settings und holt den Pfad daraus.